src/build_queue.rs

```diff
@@ -275,6 +277,33 @@ impl AsyncBuildQueue {
 
         Ok(())
     }
+
+    /// Decreases the priority of all releases currently present in the queue not matching the version passed to *at least* new_priority.
+    pub(crate) async fn deprioritize_other_releases(
+        &self,
+        name: &str,
+        latest_version: &Version,
+        new_priority: i32,
+    ) -> Result<()> {
+        let mut conn = self.db.get_async().await?;
+        sqlx::query!(
+            "UPDATE queue
+             SET priority = GREATEST(priority, $1)
+             WHERE
+                name = $2
+                AND version != $3
+                AND attempt < $4
```

Comment:
If I understood the code correctly judging by other existing queries, this is basically a check for failed builds - is my understanding correct?